### DISCUSSION TITLE
FieldSpec question

### DISCUSSION THREAD

**Discussion by dibyendumajumdar** [mjhoptics/ray-optics#74] [Q&A] [answered]
https://github.com/mjhoptics/ray-optics/blob/e55a4c199ddc738235813a08c562d18cce310064/src/rayoptics/raytr/opticalspec.py#L405

If I set a field spec as object angle, I notice here that only y coordinate is set. Is there a way to make the ray start such that it makes that angle on both x and y axes?

**Comment by mjhoptics** [comment, marked as answer]
Yes, but you have to do it field by field; there isn't a single function call to do it.

My general approach with ray-optics has been to implement general purpose algorithms in the "back-end" but only implement interface functions to handle the most common systems, i.e. rotationally symmetric systems. So the `set_from_list()` method of `FieldSpec` is really only accepting y or radial field heights for convenience in the most common situations. If you have an anamorphic or bilaterally symmetric system, then the user needs to add `Field` objects directly to `FieldSpec` with the appropriate `x, y` coordinates set. The documentation doesn't make this clear (which is an understatement) so thank you for asking!
